test/e2e/framework/syncer.go

```diff
@@ -525,9 +560,92 @@ type StartedSyncerFixture struct {
 
 	SyncerVirtualWorkspaceConfig   *rest.Config
 	UpsyncerVirtualWorkspaceConfig *rest.Config
+
+	stopHeartBeat context.CancelFunc
+}
+
+// StartHeartBeat starts the Heartbeat keeper to maintain
+// the SyncTarget to the Ready state.
+// No resource will be effectively synced after calling this method.
+func (sf *appliedSyncerFixture) StartHeartBeat(t *testing.T) *StartedSyncerFixture {
+	t.Helper()
+
+	ctx, cancelFunc := context.WithCancel(context.Background())
+	t.Cleanup(cancelFunc)
+	sf.stopHeartBeat = cancelFunc
+
+	kcpBootstrapClusterClient, err := kcpclientset.NewForConfig(sf.SyncerConfig.UpstreamConfig)
+	require.NoError(t, err)
+	kcpSyncTargetClient := kcpBootstrapClusterClient.Cluster(sf.SyncerConfig.SyncTargetPath)
+
+	// Start the heartbeat keeper to have the SyncTarget always ready during the e2e test.
+	syncer.StartHeartbeatKeeper(ctx, kcpSyncTargetClient, sf.SyncerConfig.SyncTargetName, sf.SyncerConfig.SyncTargetUID)
+
+	startedSyncer := &StartedSyncerFixture{
+		sf,
+	}
+
+	// The sync target becoming ready indicates the syncer is healthy and has
+	// successfully sent a heartbeat to kcp.
+	startedSyncer.WaitForClusterReady(ctx, t)
```

Comment:
cluster?

Reply:
This is the existing name of the method. I can change it if you prefer, of course, to `WaitForSyncTargetReady`

Reply:
Fixed in commit 13a4a065c80f0bf3815d69defc0ffcd0728fce2d